scene/gui/progress_bar.cpp

```diff
@@ -85,6 +123,16 @@ void ProgressBar::_notification(int p_what) {
 	}
 }
 
+void ProgressBar::set_fill_mode(int p_fill) {
+	ERR_FAIL_INDEX(p_fill, 9);
```

Comment:
Actually I think hardcoding this is bad practice. It is very easy to miss in the future if more items are added to the enum. Using the `MAX` last enum element is used very often in the codebase.

```suggestion
	ERR_FAIL_INDEX(p_fill, FILL_MODE_MAX);
```